Since musl 1.2 time_t is a 64 bit value, even on 32 bit systems. A
hotfix for printing the value is simply using PRIu64 from inttypes.h
in the format string.

diff -upr openrc-0.42.1.orig/src/rc/rc-status.c openrc-0.42.1/src/rc/rc-status.c
--- openrc-0.42.1.orig/src/rc/rc-status.c	2021-01-05 18:21:13.841847588 +0100
+++ openrc-0.42.1/src/rc/rc-status.c	2021-01-05 18:23:11.682103438 +0100
@@ -20,6 +20,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
+#include <inttypes.h>
 
 #include "einfo.h"
 #include "queue.h"
@@ -128,12 +129,12 @@ static char *get_uptime(const char *serv
 			}
 			if (diff_days > 0)
 				xasprintf(&uptime,
-						"%ld day(s) %02ld:%02ld:%02ld (%s)",
+						"%"PRIu64" day(s) %02"PRIu64":%02"PRIu64":%02"PRIu64" (%s)",
 						diff_days, diff_hours, diff_mins, diff_secs,
 						start_count);
 			else
 				xasprintf(&uptime,
-						"%02ld:%02ld:%02ld (%s)",
+						"%02"PRIu64":%02"PRIu64":%02"PRIu64" (%s)",
 						diff_hours, diff_mins, diff_secs, start_count);
 		}
 	}
